Data race during restore

Travis caught a data race during restore:

```
==================
WARNING: DATA RACE
Write at 0x00c4238a6748 by goroutine 545:
  github.com/hashicorp/vault/vault.TestTokenStore_CreateLookup_ExpirationInRestoreMode()
      /home/travis/gopath/src/github.com/hashicorp/vault/vault/token_store_test.go:569 +0xcf7
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c

Previous read at 0x00c4238a6748 by goroutine 611:
  github.com/hashicorp/vault/vault.(*ExpirationManager).Restore.func2()
      /home/travis/gopath/src/github.com/hashicorp/vault/vault/expiration.go:338 +0x99

Goroutine 545 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:921 +0x206
  main.main()
      github.com/hashicorp/vault/vault/_test/_testmain.go:632 +0x1d3

Goroutine 611 (finished) created at:
  github.com/hashicorp/vault/vault.(*ExpirationManager).Restore()
      /home/travis/gopath/src/github.com/hashicorp/vault/vault/expiration.go:314 +0x407
==================
--- FAIL: TestTokenStore_CreateLookup_ExpirationInRestoreMode (0.15s)
	testing.go:699: race detected during execution of test
```

--- a/vault/token_store_test.go
+++ b/vault/token_store_test.go
@@ -566,7 +566,6 @@ func TestTokenStore_CreateLookup_ExpirationInRestoreMode(t *testing.T) {
 	ts.expiration.restoreModeLock.Lock()
 	ts.expiration.restoreMode = 1
 	ts.expiration.restoreLocks = locksutil.CreateLocks()
-	ts.expiration.quitCh = make(chan struct{})
 	ts.expiration.restoreModeLock.Unlock()
 
 	// Test that the token lookup does not return the token entry due to the

